src/python/pants/util/dirutil.py

```diff
@@ -100,18 +100,34 @@ def safe_mkdir_for_all(paths):
       created_dirs.add(dir_to_make)
 
 
-def safe_file_dump(filename, payload, binary_mode=True):
+def safe_file_dump(filename, payload, binary_mode=True, mode='w'):
```

Comment:
Thinking along the same lines as [this comment from my last PR that you've already read about whether append is atomic](https://github.com/pantsbuild/pants/pull/6533#discussion_r219380141), I think calling this "safe" makes sense if we're always truncating the file with `'w'` or `'wb'`, but I'm not sure it makes sense adding append into the picture. If there's a link that shows this is definitely atomic in the case of multiple processes I'd like it to be commented here, or if that can't be done, I don't know if I'd want to extend this method with a mode argument, and if the method of assuring atomicity for appends requires extra work, I'd probably want it in a separate method anyway.

Reply:
I did not remove the `TODO` that references atomicity: using `safe_file_open` was not intended to address atomicity.

The extent to which `safe_file_open` is "safe" is explained in the `safe_open` method. I can reference it from the docstring.

Reply:
Yep, ok, looked at `safe_open` and I think no change is needed here. I would like to not make it easy to append to a file in a non-atomic way, which isn't a concern shared with `'r'` or `'w'` modes -- but I think the better way is probably to lift that out into a separate method or something.